server/src/main/java/org/elasticsearch/indices/breaker/HierarchyCircuitBreakerService.java

```diff
@@ -251,7 +251,15 @@ private ParentMemoryUsage parentUsed(long newBytesReserved) {
 
     //package private to allow overriding it in tests
     long currentMemoryUsage() {
-        return MEMORY_MX_BEAN.getHeapMemoryUsage().getUsed();
+        try {
+            return MEMORY_MX_BEAN.getHeapMemoryUsage().getUsed();
+        } catch (IllegalArgumentException ex) {
```

Comment:
Neither of the methods here document that they can throw an `IllegalArgumentException`. So this catch looks quite odd and now I wonder if it's too broad. Catching it is fine, but can also check the exception message and `assert` that it is what we expect it to be? I don't want us to end up suppressing anything else here.

Reply:
The `IllegalArgumentException` is thrown by the  [constructor of `MemoryUsage`](http://hg.openjdk.java.net/jdk/jdk11/file/1ddf9a99e4ad/src/java.management/share/classes/java/lang/management/MemoryUsage.java#l165). It does contain a few more precondition checks but I can add an assert. However, this we need to resort to pattern matching as the message contains absolute numbers, e.g:

```
committed = 542113792 should be < max = 536870912
```

Reply:
Ah ok. Then I just misunderstood. Anyway, I can implement the assertion as suggested.